### PR TITLE
factor out conditional LanguageSelector as functional component

### PR DESCRIPTION
**Pull request by t3chguy**
delint and run generate-eslint-error-ignore-file to prevent lint regressions

Removes `returnToAppJsx` which was unset in all cases as was commented since ILAG. It has been a while since ILAG so I opted to remove the commented out code.

### Fixes https://github.com/vector-im/riot-web/issues/6901

### CODE REVIEW

File: src/components/structures/login/LanguageSelector.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import React from 'react';
+
+function onChange(newLang) {
+    if (getCurrentLanguage() !== newLang) {
+        SettingsStore.setValue("language", null, SettingLevel.DEVICE, newLang);
+        PlatformPeg.get().reload();
+    }
+}
+
+export default function makeLanguageSelector() {
```
Comment (lukebarnard1) [inline src/components/structures/login/LanguageSelector.js]: Typically we would export something like `function LanguageSelector()`

File: src/components/structures/login/Login.js [hunk 2]
```diff
@@ -461,7 +444,7 @@ module.exports = React.createClass({
                         </a>
                         { loginAsGuestJsx }
                         { returnToAppJsx }
-                        { !SdkConfig.get().disable_login_language_selector ? this._renderLanguageSetting() : '' }
+                        { makeLanguageSelector() }
```
Comment (lukebarnard1) [inline src/components/structures/login/Login.js]: Normally we would use JSX, i.e. `<LanguageSelector />`

File: src/components/structures/login/Login.js [hunk 3]
```diff
@@ -20,12 +21,11 @@ limitations under the License.
...
 import PropTypes from 'prop-types';
 import { _t } from '../../../languageHandler';
-import * as languageHandler from '../../../languageHandler';
 import sdk from '../../../index';
 import Login from '../../../Login';
-import PlatformPeg from '../../../PlatformPeg';
 import SdkConfig from '../../../SdkConfig';
-import SettingsStore, {SettingLevel} from "../../../settings/SettingsStore";
+import SettingsStore from "../../../settings/SettingsStore";
+import makeLanguageSelector from "./LanguageSelector";
```
Comment (lukebarnard1) [inline src/components/structures/login/Login.js]: Instead of importing this directly, we should be using `sdk.getComponent`

Comment (t3chguy) [reply src/components/structures/login/Login.js]: Fair enough

Re this there's actually more instances of direct imports of `<AccessibleButton />` than skinned imports.
29 vs 16 :(

Comment (lukebarnard1) [reply src/components/structures/login/Login.js]: >there's actually more instances of direct imports of <AccessibleButton /> than skinned imports.

Yeah, for some reason it's almost systematically been done incorrectly for AccessibleButton. 

It's unfortunate that we're paying the price for the component drop-in system without any benefits.

File: src/components/structures/login/Registration.js [hunk 4]
```diff
@@ -64,6 +63,12 @@ module.exports = React.createClass({
         onLoginClick: PropTypes.func.isRequired,
         onCancelClick: PropTypes.func,
         onServerConfigChange: PropTypes.func.isRequired,
+
+        rtsClient: PropTypes.shape({
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/components/structures/login/Registration.js]: Except then it'll fail during testing when it tries to pass in an object? https://github.com/matrix-org/matrix-react-sdk/blob/bf0ec249b686ac89d4344fd654ffd9bc5cfbef62/test/test-utils.js#L106

[the remark this answers is not in the document]
Comment (lukebarnard1) [reply src/components/structures/login/Registration.js]: oh yay, mocking :cry: 

Fair enough, @t3chguy :+1:

**Pull request merged**